server/spec/consumer_resource_host_guest_spec.rb

```diff
@@ -195,6 +195,25 @@
     host2['uuid'].should == host_consumer2['uuid']
   end
 
+  it 'should ignore duplicate guest ids' do
+    uuid1 = random_string('system.uuid')
+    uuid2 = random_string('system.uuid')
+    guests = [{'guestId' => uuid1}, {'guestId' => uuid1}, {'guestId' => uuid2},  {'guestId' => uuid2}]
+
+    user_cp = user_client(@owner1, random_string('test-user'))
+    host_consumer = user_cp.register(random_string('host'), :system, nil,
+      {}, nil, nil, [], [])
+    user_cp.register(random_string('guest'), :system, nil,
+      {'virt.uuid' => uuid1}, nil, nil, [], [])
+    user_cp.register(random_string('guest'), :system, nil,
+      {'virt.uuid' => uuid2}, nil, nil, [], [])
+
+    consumer_client = Candlepin.new(nil, nil, host_consumer['idCert']['cert'], host_consumer['idCert']['key'])
+    consumer_client.update_consumer({:guestIds => guests})
+
+    @cp.get_consumer_guests(host_consumer['uuid']).length.should == 2
```

Comment:
Should also verify the UUIDs received are the two expected, remembering that the order isn't guaranteed.

Reply:
Nice suggestion. I have added a check for UUIDs. 

Regarding commit message, Sure, I will try to add details about the fix.